test/unified_memory_allocation/uma_pools/disjoint_pool_config.cpp

```diff
@@ -0,0 +1,325 @@
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).Capacity, 3);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).SlabMinSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for read-only shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).limits->MaxSize, 32 * 1024 * 1024);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringDisabledBuffersTest) {
+    // test for valid string with disabled buffers-- (the map should be empty)
+    std::string config = "0;32M;host:1M,4,64k;device:1m,4,64K;shared:0,3,2M";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    ASSERT_TRUE(configsMap.empty());
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringEmpty) {
+    // test for empty string-- (all the values should be default
+    // set by getConfigurationsFor)
+    std::string config = "";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    // test for host
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).MaxPoolableSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for device
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).Capacity, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for read-only shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).limits->MaxSize, 16 * 1024 * 1024);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringInvalid) {
+    // test for invalid string-- (all the values should be default
```

Comment:
Since there are several tests with the same result (all values default), use parameterized test with a config string as the parameter. See http://google.github.io/googletest/reference/testing.html#TEST_P

Reply:
Yeah, good idea, thanks. Done.